### PR TITLE
[OTel] Grpcio instrumentation

### PR DESCRIPTION
**Pull request by GSVarsha**
`coverage: 95%`
- Type annotations will be handled later

### CODE REVIEW

File: src/instana/instrumentation/grpcio.py [hunk 1]
```diff
@@ -33,248 +38,308 @@ def collect_tags(span, instance, argv, kwargs):
...
+                    disable_w3c_trace_context=True,
+                )
+                collect_attributes(span, instance, argv, kwargs)
+                span.set_attribute("rpc.call_type", "unary")
 
                 rv = wrapped(*argv, **kwargs)
             except Exception as e:
-                scope.span.log_exception(e)
+                span.record_exception(e)
                 raise
```
Comment (pvital) [inline src/instana/instrumentation/grpcio.py:82]: You can skip this `except` block from the coverage (that it's unnecessary to check) and reduce the number of error spans logged if you use the following:

```
    except Exception as exc:
        span.record_exception(exc)
else:
```

You can apply this for the rest of the code (and other files).

**Review by pvital — CHANGES_REQUESTED** [review]
One request only.

File: src/instana/instrumentation/grpcio.py (hunk 1 above)
Comment (GSVarsha) [reply src/instana/instrumentation/grpcio.py]: Done.

**Review by pvital — APPROVED** [review]
It looks good to me.

**Pull request merged**